src/ratingservice/main_test.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+import os
+import unittest
+import requests
+import json
+
+class TestEndpoints(unittest.TestCase):    
+    def testGetRating(self):
+        """ Test if getting the rating of a product returns success """
+        products = read_products()
+        for product in products:
+            url = "https://ratingservice-dot-{0}.wl.r.appspot.com/getRating/{1}".format(getProjectId(), product)
+            res = requests.get(url)
+            self.assertEqual(res.status_code, 200)
+
+    def testGetRatingNotExist(self):
+        """ Test if getting non-existing product returns 404 """
+        url = "https://ratingservice-dot-{0}.wl.r.appspot.com/getRating/{1}".format(getProjectId(), "random")
+        res = requests.get(url)
+        self.assertEqual(res.status_code, 404)
+    
+    def testRate(self):
+        """ Test if rating a product returns success """
+        products = read_products()
+        for product in products:
+            url = "https://ratingservice-dot-{}.wl.r.appspot.com/rate".format(getProjectId())
+            res = requests.post(url, data={
+                'score' : 5,
+                'id' : product
+            })
+            self.assertEqual(res.status_code, 200)
+    
+    def testGetRatingCorrect(self):
+        """ Test if getting the rating of a product returns a correct number """
+        url_get = "https://ratingservice-dot-{0}.wl.r.appspot.com/getRating/{1}".format(getProjectId(), "OLJCESPC7Z")
+        url_post = "https://ratingservice-dot-{0}.wl.r.appspot.com/rate".format(getProjectId())
+        res1 = requests.get(url_get).json()
+        requests.post(url_post, data={
+            'score' : 5,
+            'id' : 'OLJCESPC7Z'
+        })
+        res2 = requests.get(url_get).json()
+        # The original total score (count * rating) plus the current score 5 must equal to the current total score
+        self.assertTrue(abs(float(res1['rating']) * float(res1['count']) + 5 
+                        - float(res2['rating']) * float(res2['count'])) < 1e-5)
+    
+
+def getProjectId():
+    return os.environ['GOOGLE_CLOUD_PROJECT']
+
+# read product ids
+def read_products():
+    res = []
+    with open('products.json') as f:
```

Comment:
tests file should be placed with the rest of the tests we have, not in the same folder as the service code. As a result, you might need to read products.json assuming it's not on the same directory

Reply:
I changed the location.